src/routes/notFound/NotFoundPage.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+
+import React from 'react'
+import { connect } from 'react-redux'
+import Failure from '../../modules/common/components/Failure'
+import { goToI18nRedirect } from '../../modules/app/routes/i18nRedirect'
+import LanguageSelector from '../../modules/common/containers/LanguageSelector'
+import CityModel from '../../modules/endpoint/models/CityModel'
+
+type Props = {
+  city: string,
+  language?: string,
+  cities: Array<CityModel>,
+  invalidCity: ?string
+}
+
+/**
+ * Our error component, but since the name Error collides with the ES6 class, we've called it Failure
+ */
+export class NotFoundPage extends React.Component<Props> {
+  getCityName (): ?string {
+    const {city, cities} = this.props
+    const cityModel = cities.find(_city => _city.code === city)
```

Comment:
I don't really like the _ code-style. I'd prefere real names here:
`const cityModel = cities.find(city => city.code === lookupCity)`

Reply:
I won't refactor all of this in this task. Apart from that, I like it :D